src/swresample_stubs.c

```diff
@@ -327,30 +327,42 @@ static void alloc_out_planar_float_array(swr_t *swr, int out_nb_samples)
 static int convert_to_planar_float_array(swr_t *swr, uint8_t **in_data, int in_nb_samples, int out_nb_samples)
 {
   CAMLparam0();
-  CAMLlocal1(channel);
 
-  for(int i = 0; i < swr->out_nb_channels; i++) {
-    swr->out_data[i] = (uint8_t*)Field(swr->out_vector, i);
-  }
+  uint8_t **out_data;
+  double *pcm;
+  int out_linesize;
+  int i,j,ret;
 
-  // Convert the samples.
-  int ret = swr_convert(swr->context,
-			swr->out_data,
-			out_nb_samples,
-			(const uint8_t **) in_data,
-			in_nb_samples);
+  ret = av_samples_alloc_array_and_samples(&out_data, &out_linesize, swr->out_nb_channels,
+                                           out_nb_samples, swr->out_sample_fmt, 0);
 
```

Comment:
I propose to store the buffer out_data in the structure swr in order to reallocate it only if it is necessary. Do you agree?